src/DataTable/DropdownFilters.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { DropdownButton, useWindowSize, breakpoints } from '..';
+import { requiredWhen } from './utils/propTypesUtils';
+
+/** The first filter will be as an input, additional filters will be available in a dropdown.  */
+const DropdownFilters = ({
+  columns,
+}) => {
+  const { width } = useWindowSize();
+  const [breakoutFilter, otherFilters] = useMemo(() => {
+    const availableFilters = columns.filter((column) => column.canFilter);
+    if (width < breakpoints.small.minWidth) {
+      return [null, availableFilters];
+    }
+
+    return [availableFilters.shift(), availableFilters];
+  }, [columns, width]);
```

Comment:
This is assuming that we always want the first filter exposed. I think this should be configurable or composable in a future version

Reply:
https://openedx.atlassian.net/browse/PAR-273

Reply:
Oops that was for bulk actions. Gonna make one for filters too
https://openedx.atlassian.net/browse/PAR-274